cmd/updater/update.sh

```diff
@@ -330,20 +335,63 @@ function validate_update() {
     return 0
 }
 
+function is_system_service() {
+    systemctl status "algorand@$(systemd-escape "$1")" &> /dev/null
+    echo $?
+}
+
+function is_user_service() {
+    systemctl --user status "algorand@$(systemd-escape "$1")" &> /dev/null
+    echo $?
+}
+
+function run_systemd_action() {
+    if [ "$(uname)" = "Darwin" ]; then
+        return 1
+    fi
+
+    local action=$1
+    local data_dir=$2
+
+    system_service=$(is_system_service "${data_dir}")
+    if [ "$system_service" -eq 0 ]; then
+        process_owner=$(awk '{ print $1 }' <(ps aux | grep "[a]lgod -d ${data_dir}"))
+        if [ "$action" = "stop" ] && $IS_ROOT || $(grep sudo <(groups "$process_owner" &> /dev/null)); then
+            sudo -n systemctl "${action}" "algorand@$(systemd-escape "${data_dir}")" &> /dev/null
+        fi
+    elif [ "$system_service" -ne 4 ]; then
+        if [ "$action" = "start" ]; then
+            sudo -n systemctl "${action}" "algorand@$(systemd-escape "${data_dir}")" &> /dev/null
+        fi
+    else
+        user_service=$(is_user_service "$data_dir")
+        if [ "$user_service" -eq 0 ]; then
+            process_owner=$(awk '{ print $1 }' <(ps aux | grep "[a]lgod -d ${data_dir}"))
+            if [ "$action" = "stop" ] && [ "$(whoami)" = "$process_owner" ]; then
+                systemctl --user "${action}" "algorand@$(systemd-escape "${data_dir}")" &> /dev/null
+            fi
+        elif [ "$user_service" -ne 4 ]; then
+            if [ "$action" = "start" ]; then
+                systemctl --user "${action}" "algorand@$(systemd-escape "${data_dir}")" &> /dev/null
+            fi
+        fi
```

Comment:
Should you include an else clause for $user_service == 4 and print an error.

Reply:
Good catch, it should return a non-zero exit code so the startup code will start with `goal`.

Reply:
Thinking about it, I may re-work the caller a bit.  It seems non-intuitive to return an error here since a service doesn't have to be installed.

Reply:
I'm going to resolve this conversation since this particular code has been refactored and the comments are outdated.